corehq/apps/reports/commtrack/ledgers_by_location.py

```diff
@@ -0,0 +1,121 @@
+        Location 2 |       132 |        49 |
+    """
+
+    def __init__(self, domain, section_id, page_start, page_size):
+        self.domain = domain
+        self.section_id = section_id
+        self.page_start = page_start
+        self.page_size = page_size
+
+    @property
+    @memoized
+    def products(self):
+        if SQLProduct.objects.filter(domain=self.domain).count() > 20:
+            raise Http400("This domain has too many products.")
+        return list(SQLProduct.objects.filter(domain=self.domain).order_by('name'))
+
+    def get_locations_queryset(self):
+        return (SQLLocation.objects
+                .filter(domain=self.domain)
+                .order_by('name'))
+
+    @property
+    @memoized
+    def location_rows(self):
+        def get_location_ledger(location):
+            stock = (StockState.objects
+                     .filter(section_id=self.section_id,
+                             sql_location=location)
+                     .values_list('sql_product__product_id', 'stock_on_hand'))
+            return LocationLedger(
+                location,
+                {product_id: soh for product_id, soh in stock}
+            )
+
+        start, stop = self.page_start, self.page_start + self.page_size
+        locations = self.get_locations_queryset()[start:stop]
+        return map(get_location_ledger, locations)
+
+    @property
+    def rows(self):
+        for ledger in self.location_rows:
+            yield [ledger.location.name] + [
+                ledger.stock.get(p.product_id, 0) for p in self.products
+            ]
+
+    @property
+    def headers(self):
+        return [_("Location")] + [p.name for p in self.products]
+
+    @property
+    @memoized
+    def total_locations(self):
+        return self.get_locations_queryset().count()
+
+
+class LedgersByLocationReport(GenericTabularReport, CommtrackReportMixin):
+    name = ugettext_lazy('Ledgers By Location')
+    slug = 'ledgers_by_location'
+    ajax_pagination = True
+    # TODO actually filter by these
+    fields = [
+        'corehq.apps.reports.filters.fixtures.AsyncLocationFilter',
+        'corehq.apps.reports.dont_use.fields.SelectProgramField',
+    ]
+
+    @method_decorator(SUPPLY_REPORTS.required_decorator())
+    def dispatch(self, *args, **kwargs):
+        return super(LedgersByLocationReport, self).dispatch(*args, **kwargs)
+
+    @staticmethod
+    def show_in_navigation(domain, project, user=None):
+        return project.commtrack_enabled and SUPPLY_REPORTS.enabled(domain)
+
+    @property
+    @memoized
+    def data(self):
+        return LedgersByLocationDataSource(
+            domain=self.domain,
+            section_id=STOCK_SECTION_TYPE,
+            page_start=self.pagination.start,
+            page_size=self.pagination.count,
+        )
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            *[DataTablesColumn(header) for header in self.data.headers]
```

Comment:
I intend to add sorting eventually, so I left that in.  Probably best to remove it 'till it does something though.